service/controller/resource/app/current.go

```diff
@@ -19,11 +20,13 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) ([]*v1a
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
 
-	// The app custom resource is deleted implicitly by the provider operator
-	// when it deletes the tenant cluster namespace in the control plane.
-	if key.IsDeleted(&cr) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("not deleting apps for tenant cluster %#q", key.ClusterID(&cr)))
+	if cc.Status.Endpoint.Base == "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "no endpoint base in controller context yet")
```

Comment:
Both the kubeconfig and clusterconfigmap resources have this logic to cancel the resource until the endpoint is available.

It should also be in the app resource because we should not create app CRs until the cluster configmaps have been created.